scripts/src/javatests/org/oppia/android/scripts/todo/ClosedIssueCheckTest.kt

```diff
@@ -0,0 +1,227 @@
+    val testContent1 =
+      """
+      // TODO(#169877): test description 1
+      
+      // TODO(#1021211): test description 2
+      """.trimIndent()
+    val testContent2 =
+      """
+      # test comment 3
+      # test todo which is not a todo
+      # TODO(#169877): test description 3
+      """.trimIndent()
+    val testContent3 =
+      """
+      <!-- TODO(#169877): test description 4 -->
+      
+      <!-- TODO(#174144): test description 5 -->
+      <!-- TODO(#169877): test description 6 -->
+      """.trimIndent()
+    tempFile1.writeText(testContent1)
+    tempFile2.writeText(testContent2)
+    tempFile3.writeText(testContent3)
+
+    val exception = assertThrows(Exception::class) {
+      main(retrieveTestFilesDirectoryPath(), "169877", "abmzuyt")
+    }
+
+    assertThat(exception).hasMessageThat().contains(CLOSED_ISSUE_CHECK_FAILED_OUTPUT_INDICATOR)
+    val failureMessage =
+      """
+      The following TODOs are unresolved for the closed issue:
+      - ${retrieveTestFilesDirectoryPath()}/TempFile1.kt:1
+      - ${retrieveTestFilesDirectoryPath()}/TempFile2.bazel:3
+      - ${retrieveTestFilesDirectoryPath()}/TempFile3.xml:1
+      - ${retrieveTestFilesDirectoryPath()}/TempFile3.xml:4
+      """.trimIndent()
+    assertThat(outContent.toString().trim()).isEqualTo(failureMessage)
+  }
+
+  @Test
+  fun testClosedIssueCheck_triggerFailure_generatedFileContentShouldContainFailureTodoList() {
+    val tempFile3 = tempFolder.newFile("testfiles/TempFile3.xml")
+    val tempFile2 = tempFolder.newFile("testfiles/TempFile2.bazel")
+    val tempFile1 = tempFolder.newFile("testfiles/TempFile1.kt")
+    val testContent1 =
+      """
+      // TODO(#169877): test description 1
+      
+      // TODO(#1021211): test description 2
+      """.trimIndent()
+    val testContent2 =
+      """
+      # test comment 3
+      # test todo which is not a todo
+      # TODO(#169877): test description 3
+      """.trimIndent()
+    val testContent3 =
+      """
+      <!-- TODO(#169877): test description 4 -->
+      
+      <!-- TODO(#174144): test description 5 -->
+      <!-- TODO(#169877): test description 6 -->
+      """.trimIndent()
+    tempFile1.writeText(testContent1)
+    tempFile2.writeText(testContent2)
+    tempFile3.writeText(testContent3)
+
+    val exception = assertThrows(Exception::class) {
+      main(retrieveTestFilesDirectoryPath(), "169877", "abmzuyt")
+    }
+    val fileContentList =
+      File("${retrieveTestFilesDirectoryPath()}/todo_list.txt").useLines { it.toList() }
+
+    assertThat(exception).hasMessageThat().contains(CLOSED_ISSUE_CHECK_FAILED_OUTPUT_INDICATOR)
+    val failureMessage =
+      """
+      The following TODOs are unresolved for the closed issue:
+      - ${retrieveTestFilesDirectoryPath()}/TempFile1.kt:1
+      - ${retrieveTestFilesDirectoryPath()}/TempFile2.bazel:3
+      - ${retrieveTestFilesDirectoryPath()}/TempFile3.xml:1
+      - ${retrieveTestFilesDirectoryPath()}/TempFile3.xml:4
```

Comment:
This already seems verified in the previous test--suggest omitting it here & focusing on the outcome specific to the behavior corresponding to this test.

Reply:
Done.